sssd_test_framework/utils/adcli.py

```diff
@@ -0,0 +1,198 @@
+
+        :param domain: Domain.
+        :type domain: str
+        :param args: Additional arguments, defaults to None
+        :type args: list[str] | None, optional
+        :return: Result of called command.
+        :rtype: ProcessResult
+        """
+        if args is None:
+            args = []
+
+        return self.host.conn.exec(["adcli", "testjoin", "--verbose", domain, *args], raise_on_error=False)
+
+    def join(
+        self,
+        domain: str,
+        *,
+        args: list[str] | None = None,
+        password: str,
+        login_user: str,
+        krb: bool = False,
+    ) -> ProcessResult:
+        """
+        Create a computer account.
+
+        :param domain: Domain.
+        :type domain: str
+        :param args: Additional arguments, defaults to None
+        :type args: list[str] | None, optional
+        :param password: Password
+        :type password: str
+        :param login_user: Authenticating User
+        :type login_user: str
+        :param krb: Use Kerberos credentials, defaults to False
+        :type krb: bool, optional
+        :return: Result of called command.
+        :rtype: ProcessResult
+        """
+        if args is None:
+            args = []
+
+        if krb:
+            self.host.conn.exec(["kinit", f"{login_user}@{domain.upper()}"], input=password)
+            command = self.host.conn.exec(["adcli", "join", "--verbose", "-C", *args, domain], raise_on_error=False)
+        else:
+            command = self.host.conn.exec(
+                ["adcli", "join", "--stdin-password", "--verbose", f"--login-user={login_user}", *args, domain],
+                input=password,
+                raise_on_error=False,
+            )
+
+        return command
+
+    def delete_computer(
+        self,
+        domain: str,
+        *,
+        args: list[str] | None = None,
+        password: str,
+        login_user: str,
+        krb: bool = False,
+    ) -> ProcessResult:
+        """
+        Delete computer account.
+
+        :param domain: Domain.
+        :type domain: str
+        :param args: additional arguments, defaults to None.
+        :type args: list[str] | None, optional
+        :param password: Password
+        :type password: str
+        :param login_user: Authenticating User
+        :type login_user: str
+        :param krb: Use Kerberos credentials, defaults to False
+        :type krb: bool, optional
+        :return: Result of called command.
+        :rtype: ProcessResult
+        """
+        if args is None:
+            args = []
+
+        if krb:
+            self.host.conn.exec(["kinit", f"{login_user}@{domain.upper()}"], input=password)
+            command = self.host.conn.exec(["adcli", "delete-computer", "--verbose", "-C", *args], raise_on_error=False)
+        else:
+            command = self.host.conn.exec(
+                ["adcli", "delete-computer", "--stdin-password", "--verbose", *args], input=password, raise_on_error=False
```

Comment:
I think there's a linter error about this line being too long.   You might want to re-run `black` against it to let it reformat to clear up that error.